pkg/util/arith.go

```diff
@@ -34,6 +34,15 @@ func Min[T constraints.Ordered](x, y T) T {
 	}
 }
 
+// AbsDiff returns the absolute value of the difference between x and y
+func AbsDiff[T constraints.Unsigned](x, y T) T {
+	if x > y {
+		return x - y
+	} else {
```

Comment:
you don't need the else because you early return

Reply:
Keeping as-is for symmetry and consistency with other parts in the file.